### ISSUE TITLE
Send email requesting signature after setting visible signatures (v5.1.4)

### ISSUE THREAD

**Issue by tasagore** [LibreSign/libresign#886]
Libresign v5.1.4 and NC24

1) From any file click on the details (three dots at the right)
2) In the LibreSign tab click on "Request"
3) Fill the email and click in "Add"
4) Click on "Submit Request"
5) A popup ask if we want to configure visible signatures, should be a YES/NO dialog but is a OK/CANCEL dialog

And now two behaviors, the first is right:

6) Click "Cancel"
7) The user receives a email asking for the signature and the file can be signed

But:

6) Click "Ok"
7) The user receives a email asking for the signature and the file can be signed BUT it's still in draft mode so he can't sign the document. If we setup the visible signatures then he will be able to sign but it we don't setup that the document will remain in draft mode and the user will see a "document not available" message if he tries to sign it.

Expected behavior: the email should be sent after then visible signature setup, not before. In fact the user should not receive any email if the document is in draft mode.

**Comment by vitormattos** [comment]
In the step 7 you will see a screen with preview of all pages of document and a list with all that need to sign.
Choose the page that you want to put a visible signature
Click in person that you want to sign in this page
Fix the position and size of visible signature
Click in save
Do this for all and after this, you will need to send the document to sign, changing the status of document from draft to be able to sign.

I think that the users received the email because you send the document before the flow that I described above.

**Comment by tasagore** [comment]
> In the step 7 you will see a screen with preview of all pages of document and a list with all that need to sign. Choose the page that you want to put a visible signature Click in person that you want to sign in this page Fix the position and size of visible signature Click in save Do this for all and after this, you will need to send the document to sign, changing the status of document from draft to be able to sign.
> 
> I think that the users received the email because you send the document before the flow that I described above.

Hi

The problem is that the app sends the email after step 6, no matters if you complete or not step 7 (put visible signatures in place). 

The email should not be sent until the document is finally ready to be signed (after the visible signatures are in place or when the requester validate the document).

I mean, it has no sense to send a email asking for a signature in a document that is not ready to be signed.

**Comment by vitormattos** [comment]
I think that this issue is outdated and this already was solved.

Check the newest version of LibreSign. If you found anything with strange behavior, create a new issue or if is related to this issue, put the #886 code to make reference to this issue.
